add function generateThumbnail

mixcas opened the pull request:
generate thumbnails when an image is uploaded

Commit:
add function generateThumbnail: generate thumbnails when an image is uploaded

varffs commented:
I think we should add quite a lot of sizes me might want at this stage rather than later as we dont want to have to work out how to regenerate image sizes on already uploaded images. Better to have too many sizes than not enough

varffs commented:
How do we code review stuff like this? I can't see any way to actually test this code (how are you writing it even?)

varffs commented:
It looks fine tho except the sizes thing

mixcas commented:
@varffs i hate it too cuz i'm working this and deploying it, testing it and realtime and i hate it. I couldn't make it work locally. I'm not sure how to but there must be a way.

I'll appreciate suggestions for the sizes.

mikeyrayvon commented:
i suggest we derive sizes from device widths. 

375, 768, 1024 are iphone, ipad and ipad pro. retina so 750, 1536, 2048.

and 300 works for admin thumb and grid view on iphone.

so our sizes array could be `[300, 750, 1536, 2048]`

what say you?

varffs commented:
I think we should ad a mid size like 1024 or something like that and then it looks good. We should probably also add a really small 50 or 100 or something that might be useful. Are we building in the ability to square crop at all?

mixcas commented:
Yeah, a 300x300 hard crop would be good.

I'm gonna do all those changes. I also wanna clean the code a bit.

Commit:
clean generateThumbnail with better image sizes declaration. add local run file

mixcas commented:
Added the suggested sizes and a better declaration. For each of the suggested sizes we generate resized images of given sizes and squared cropped images of width and height of given sizes.

Ex, `350`, generates a 350px max width image and a 350x350px image.

Updated readme

Commit:
update readme

varffs commented on functions/test/generateThumbnail.js:
```diff
@@ -0,0 +1,26 @@
+  updated:'2018-02-01T20:39:15.477Z'
+};
+
+generateThumbnail(data);
```
should this test run some kinda of assertation on the return value?

mixcas replied:
Idk what it returns. Would be good to check

varffs commented:
Read the code it looks good. Ran the test it worked.
